dsl_lang/lib/ast.ml

```diff
@@ -0,0 +1,10 @@
+type clss = string
+
+type var = string
+
+type policy =
+| Class of clss
```

Comment:
This is great for now, but I think you have essentially made a constructor to accommodate syntactic sugar. Recall that a single class like "C" written out as a policy is meant to be desugared into `Fifo [C]`.

My hope is that eventually we won't need this constructor `Class of clss`, and instead, if the user gives us a single class `C` when we are expecting a policy, we will automatically parse it into the policy `Fifo [C]`. That is, we'll use the `Fifo` constructor with a singleton list containing `C`. Let me know if you'd like to chat more about this!

Reply:
I think I almost understand what you mean, but not quite. It would be good to chat!

Reply:
Sorry I missed this! I didn’t see you in the lab so I left you a note on the whiteboard! But we can also do a call tomorrow :)

Reply:
But, yes, if the constructor `Fifo` takes a `policy list`, then you’ll need a constructor `Class of clss`. Sorry if that was the cause of confusion. So you’ll have the constructor around, but you’ll never really parse a user-given policy into that constructor. That constructor will mostly be used as an ingredient for other constructors.

Reply:
Ok, I see what you're getting at (also got your whiteboard note). Not exactly sure how I'll implement it yet, but I guess I'll cross that bridge when I get there.

Reply:
Now that I'm trying to implement it, the part that confuses me is knowing when to parse a capital letter as a class vs as sugar for fifo. Maybe having the sugar as `[A]` instead of `A` could help, or something similar to distinguish them? Or is there another way that I'm missing?